queries.sql

```diff
@@ -188,4 +188,152 @@ FROM animals A
 JOIN owners O
 ON A.owner_id = O.id
 GROUP BY O.full_name
-ORDER BY animal DESC;
\ No newline at end of file
+ORDER BY animal DESC;
+
+/*MANY-TO-MANY QUERIES*/
+
+/*"Who was the last animal seen by William Tatcher?"*/
+
+SELECT TOP 1
+A.name AS animal_name,
+vets.name AS vet_name,
+visits.date_of_visit
+FROM visits
+JOIN animals A
+ON A.id = visits.animal_id
+JOIN vets
+ON vets.id = visits.vet_id
+WHERE vets.name = 'William Tatcher'
+ORDER BY visits.date_of_visit DESC;
+
+/*"How many different animals did Stephanie Mendez see?"*/
+
+SELECT DISTINCT
+vets.name,
+COUNT(DISTINCT A.name) AS animals_attended
+FROM visits
+JOIN animals A
+ON A.id = visits.animal_id
+JOIN vets
+ON vets.id = visits.vet_id
+WHERE vets.name = 'Stephanie Mendez'
+GROUP BY vets.name;
+
+/*"List all vets and their specialties,
+including vets with no specialties."*/
+
+SELECT
+vets.name AS vet_name,
+species.name AS specialization
+FROM specializations S
+JOIN species
+ON species.id = S.species_id
+RIGHT JOIN vets
+ON vets.id = S.vet_id;
+
+/*"List all animals that visited Stephanie Mendez between
+April 1st and August 30th, 2020."*/
+
+SELECT
+A.name AS animal_name,
+vets.name AS vet_name,
+visits.date_of_visit
+FROM visits
+JOIN animals A
+ON A.id = visits.animal_id
+JOIN vets
+ON vets.id = visits.vet_id
+WHERE visits.date_of_visit >= '2020-04-01'
+AND visits.date_of_visit <= '2020-08-30'
+AND vets.name = 'Stephanie Mendez';
+
+/*"What animal has the most visits to vets?"*/
+
+SELECT
+A.name,
+COUNT(A.name) AS visits
+FROM visits
+JOIN animals A
+ON A.id = visits.animal_id
+GROUP BY A.name
+ORDER BY visits DESC
+LIMIT 1;
+
+/*"Who was Maisy Smith's first visit?"*/
+
+SELECT
+A.name AS animal_name,
+vets.name AS vet_name,
+visits.date_of_visit
+FROM visits
+JOIN animals A
+ON A.id = visits.animal_id
+JOIN vets
+ON vets.id = visits.vet_id
+WHERE vets.name = 'Maisy Smith'
+ORDER BY visits.date_of_visit ASC
+LIMIT 1;
+
```

Comment:
- [ ] I would love to think that you forgot to update this screenshot after working on the query. Kindly run it and update your query. The previous reviewer did well to remind you that it should return only one but this isn't as required yet.

![image](https://github.com/Movgto/vet-clinic-database/assets/99715640/eeaac885-0c1f-4ea3-9ca5-fe620fec3396)